engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -155,10 +158,11 @@ public ValidatorImpl(ConstraintValidatorFactory constraintValidatorFactory,
 		}
 
 		ValidationOrder validationOrder = determineGroupValidationOrder( groups );
-		ValueContext<?, Object> valueContext = ValueContext.getLocalExecutionContext(
+		BeanValueContext<?, Object> valueContext = ValueContexts.getLocalExecutionContextForBean(
 				validatorScopedContext.getParameterNameProvider(),
 				object,
 				validationContext.getRootBeanMetaData(),
+				validationContext.getRootBeanMetaData(),
```

Comment:
I think that there was a place where `validatable != the same metadata` so that's why I've added that other parameter. but it was pretty late yesterday so I might be wrong :smiley: I'll check this.

Reply:
From what I can see, it's always the same value right now. So if they should be different at one place, we have a bug :)